### PR TITLE
docs: Update Salesforce instructions

### PR DESCRIPTION
**Pull request by froboy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,11 +56,29 @@ The main module itself provides only API that helps fetch data from TractionRec.
...
+     - Under **Custom Object Permissions**, add **Read** access to the [Objects listed below](#salesforce-permissions).
+   - Save those changes.
+4. Create a new **User** with the new Profile:
+   - **Setup** > **Users** > **New User**
+     - **User License** - Salesforce Platform
+     - **Email** - A working email that you will use to receive login verifications.
+     - **Username** - This is _not_ your email and _must_ be unique across _all Salesforce Organizations_. **This is the name that will be used in the Drupal connection below.** If you enter a preexisting username, you will receive this error:
+        > Error: Duplicate Username.
+        > The username already exists in this or another Salesforce organization. Usernames must be unique across all Salesforce organizations. To resolve, use a different username (it doesn't need to match the user's email address).
+     - Assign the User to the **Profile** you created above, or a **Permission Set** that has the [necessary permissions](#salesforce-permissions).
```
Comment (anpolimus) [inline README.md:75]: Can we add a comment, that you have to create a new user always to assign it to the profile?

File: README.md [hunk 2]
```diff
@@ -56,11 +56,29 @@ The main module itself provides only API that helps fetch data from TractionRec.
...
+   - **Setup** > **Users** > **Profiles** > **New**
+   - When asked what **Existing Profile** to clone from, select **Standard Platform User**.
+   - In the _very large_ configuration screen, click **Edit**, then:
+     - Under **Connected App Access**, add access to the Connected App you created above.
+     - Search for and enable the [System permissions listed below](#salesforce-permissions).
+     - Under **Custom Object Permissions**, add **Read** access to the [Objects listed below](#salesforce-permissions).
+   - Save those changes.
+4. Create a new **User** with the new Profile:
+   - **Setup** > **Users** > **New User**
+     - **User License** - Salesforce Platform
```
Comment (anpolimus) [inline README.md]: When I select it, I cant select Profile that I've created for the app.
Profile field is dependent on User License. Maybe we have to skip this rule and use default one that Traction Rec offers

**Review by anpolimus — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -56,11 +56,29 @@ The main module itself provides only API that helps fetch data from TractionRec.
...
+     - **Email** - A working email that you will use to receive login verifications.
+     - **Username** - This is _not_ your email and _must_ be unique across _all Salesforce Organizations_. **This is the name that will be used in the Drupal connection below.** If you enter a preexisting username, you will receive this error:
+        > Error: Duplicate Username.
+        > The username already exists in this or another Salesforce organization. Usernames must be unique across all Salesforce organizations. To resolve, use a different username (it doesn't need to match the user's email address).
+     - Assign the User to the **Profile** you created above, or a **Permission Set** that has the [necessary permissions](#salesforce-permissions).
+5. Confirm your **Connected App**, **Profile**, and **User** are connected:
+   - Go to **Setup** > **Connected Apps** > **Manage Connected Apps** and choose your new app. Assign the **Profile** or **Permission Set** that contains your new user if it does not already show under the relevant section.
```
Comment (anpolimus) [inline README.md]:
```suggestion
   - Go to **Setup** > **Apps** > **Connected Apps** > **Manage Connected Apps** and choose your new app. Assign the **Profile** or **Permission Set** that contains your new user if it does not already show under the relevant section.
```

File: README.md [hunk 4]
```diff
@@ -56,11 +56,29 @@ The main module itself provides only API that helps fetch data from TractionRec.
...
    - Save the **Consumer Key** and **Consumer Secret** for the next step.
-4. Connect your new Connected App to a User. The attached User's email will be the one used to authenticate the app.
-   - If necessary, create a user via **Setup** > **Users** > **New User**
-   - Assign the User to a **Profile** or **Permission Set** that has the [necessary permissions](#salesforce-permissions).
-   - Go to **Setup** > **Connected Apps** > **Manage Connected Apps** and choose your new app. Assign the profile or permission set that contains your new user.
-   - In the Connect App Detail, click **Edit Policies**, then under **OAuth Policies** > **Permitted Users** choose **Admin approved users are pre-authorized**, then Save.
+4. Create a **Profile** to assign permissions to your app:
+   - You **must** do this **before** creating a user. Alternatively, existing users can be assigned a **Permision Set**. Instructions TBD.
+   - **Setup** > **Users** > **Profiles** > **New**
+   - When asked what **Existing Profile** to clone from, select **Standard Platform User**.
```
Comment (froboy) [inline README.md]:
```suggestion
   - When asked what **Existing Profile** to clone from, select **Standard User** or **Standard Platform User**. Be sure to note the **User License** connected to the target profile.
```

File: README.md (hunk 2 above)
Comment (froboy) [reply README.md]:
```suggestion
     - **User License** - The option under which you created the Profile in the previous step.
```

File: README.md (hunk 1 above)
Comment (froboy) [reply README.md:75]: https://github.com/YCloudYUSA/openy_traction_rec/pull/9/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R60

>    - You **must** do this **before** creating a user. Alternatively, existing users can be assigned a **Permision Set**. Instructions TBD.

Does line 60 cover that?

**Comment by froboy** [comment]
@anpolimus pushing these for now. I'll have a new PR soon if you have more comments.

**Pull request merged**